modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -238,7 +238,10 @@ public void run() {
 
                     // The lease is expired or close to this.
                     if (lease.getExpirationTime().getPhysical() < outdatedLeaseThreshold) {
-                        ClusterNode candidate = nextLeaseHolder(entry.getValue(), null);
+                        ClusterNode candidate = nextLeaseHolder(
+                                entry.getValue(),
+                                lease.isAccepted() ? lease.getLeaseholder().name() : null
+                        );
```

Comment:
Why do we still select next lease holder every time? I thought we are going to change this in this ticket, this shouldn't be done for prolongation.

Also, the condition for meta storage invoke on prolongation is not sufficient. Invoke on prolongation should be successfull only when 
keys are equal 
AND 
leaseholder is the same
AND
lease is accepted
AND 
lease start and expiration times are the same.

Shouldn't it?

Also, invoke condition on lease granting should be revisited accordingly.

Reply:
Yes, it is true. All your conditions are covered by tests.